### PR TITLE
Add Operator selection

### PR DESCRIPTION
**Pull request by DmitryPR**
This migration is for the new client library method that was introduced
in 0.7.19 when the operators are returned as a callback instead of
handleOperators method

IOS-127

### CODE REVIEW

File: Example/Example/ViewControllers/DemoViewController.swift [hunk 1]
```diff
@@ -85,6 +103,23 @@ extension DemoViewController {
...
     }
 
+    fileprivate func requestEngagement(selectedOperator: Operator) {
+        configureInteractor(chatType: .sync)
+
+        Salemove.sharedInstance.requestEngagement(with: selectedOperator) { [unowned self] engagementRequest, error in
+            if let reason = error?.reason {
+                self.showError(message: reason)
+            } else if let request = engagementRequest {
+                guard let controller = self.statusViewController else {
```
Comment (ostap0207) [inline Example/Example/ViewControllers/DemoViewController.swift]: What is this for?

Comment (DmitryPR) [reply Example/Example/ViewControllers/DemoViewController.swift]: controller can be nil when not configured

Comment (ostap0207) [reply Example/Example/ViewControllers/DemoViewController.swift]: You configure it just few lines above

Comment (DmitryPR) [reply Example/Example/ViewControllers/DemoViewController.swift]: Doesnt matter we are inside a callback and this callback will be called after some period of time. As we specify that the status view controller is optional then we need to first check that it's present. I would avoid force unwrap here.

Comment (ostap0207) [reply Example/Example/ViewControllers/DemoViewController.swift]: Well, what happens if it's not there? The engagement request will just continue?
The engagement request should be at least canceled.

File: Example/Example/ViewControllers/DemoViewController.swift [hunk 2]
```diff
@@ -111,6 +146,22 @@ extension DemoViewController {
         present(controller, animated: true, completion: nil)
     }
 
+    fileprivate func selectOperator(operators: [Operator], completion: @escaping (Operator) -> Void) {
```
Comment (ostap0207) [inline Example/Example/ViewControllers/DemoViewController.swift]: Aren't we already selecting operators here https://github.com/salemove/ios-bundle/pull/7/files#diff-33044241b20ee33162a4d728f6328f3fL196?

Comment (ostap0207) [reply Example/Example/ViewControllers/DemoViewController.swift]: Nvm, explained in the commit message.
Should https://github.com/salemove/ios-bundle/pull/7/files#diff-33044241b20ee33162a4d728f6328f3fL196 be removed now?

**Comment by DmitryPR** [comment]
!squash

**Comment by DmitryPR** [comment]
!merge

**Pull request merged**